From 1b516ca017fa45888d4962ebf97a78987e696209 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Sun, 10 Dec 2023 10:00:28 +0100
Subject: [PATCH] lavc/vaapi_hevc: Fixup invalid references

---
 libavcodec/vaapi_hevc.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/libavcodec/vaapi_hevc.c b/libavcodec/vaapi_hevc.c
index 3bdd2dd1b8..d08fb50e37 100644
--- a/libavcodec/vaapi_hevc.c
+++ b/libavcodec/vaapi_hevc.c
@@ -100,6 +100,8 @@ static void fill_vaapi_reference_frames(const HEVCContext *h, VAPictureParameter
     const HEVCFrame *current_picture = h->ref;
     int i, j, rps_type;
 
+    const HEVCFrame *frame_good = NULL;
+
     for (i = 0, j = 0; i < FF_ARRAY_ELEMS(pp->ReferenceFrames); i++) {
         const HEVCFrame *frame = NULL;
 
@@ -113,6 +115,22 @@ static void fill_vaapi_reference_frames(const HEVCContext *h, VAPictureParameter
         init_vaapi_pic(&pp->ReferenceFrames[i]);
 
         if (frame) {
+            if (frame->sequence != HEVC_SEQUENCE_COUNTER_INVALID) {
+                frame_good = frame;
+            } else {
+                if (frame_good) {
+                    frame = frame_good;
+                } else {
+                    for (int k = 0; k < FF_ARRAY_ELEMS(h->DPB); k++) {
+                        if ((&h->DPB[k] != current_picture || h->ps.pps->pps_curr_pic_ref_enabled_flag) &&
+                            (h->DPB[k].flags & (HEVC_FRAME_FLAG_LONG_REF | HEVC_FRAME_FLAG_SHORT_REF)) &&
+                            h->DPB[k].sequence != HEVC_SEQUENCE_COUNTER_INVALID) {
+                            frame = &h->DPB[k];
+                            break;
+                        }
+                    }
+                }
+            }
             rps_type = find_frame_rps_type(h, frame);
             fill_vaapi_pic(&pp->ReferenceFrames[i], frame, rps_type);
         }
-- 
2.43.0

